internal/workflow.go

```diff
@@ -734,6 +734,14 @@ func (wc *workflowEnvironmentInterceptor) ExecuteChildWorkflow(ctx Context, chil
 		decodeFutureImpl: mainFuture.(*decodeFutureImpl),
 		executionFuture:  executionFuture.(*futureImpl),
 	}
+
+	// Immediately return if the context is already cancelled without spawning the child workflow
+	if ctx.Err() == ErrCanceled {
```

Comment:
would check not nil be enough? I think we don't want schedule child workflow when ctx.Err() return any error.

Reply:
That's a good point. Changed.